### DISCUSSION TITLE
Estimation of the SCR peak delay

### DISCUSSION THREAD

**Discussion by BaggioMarco** [neuropsychology/NeuroKit#475] [Q&A] [answered]
Hi,
first of all thank you so much for your package!
Reading some paper I saw that scr peaks usually occur between 1-5 seconds after the onset of emotional stimuli.
Is it possible in some way to estimate such delay for a specific person?
I could not find any method or external reference on this, and I was wondering if you could share some ideas on this possibility.
thank you in advance

**Comment by zen-juen** [comment, marked as answer]
Hi @BaggioMarco! Apologies for the late response, we must have missed this by accident 😢 You can do this quite quickly with neurokit, here's how using one of our example datasets. 

```
# Example with real data
data = nk.data("bio_eventrelated_100hz")

# Process the data
df, info = nk.bio_process(eda=data["EDA"], sampling_rate=100)
events = nk.events_find(data["Photosensor"], threshold_keep='below',
                                       event_conditions=["Negative", "Neutral", "Neutral", "Negative"])
epochs = nk.epochs_create(df, events, sampling_rate=100, epochs_start=-0.1, epochs_end=6.9)

# Analyze
features = nk.eda_eventrelated(epochs)
```

The `SCR_Peak_Amplitude_Time` then gives the timepoint of each first SCR peak amplitude associated with each condition i.e., the emotional stimuli. If I'm not wrong, is this what you're specifically looking for?

```
features[features.columns[0:7]]
Out[13]: 
  Label Condition  ...  SCR_Peak_Amplitude  SCR_Peak_Amplitude_Time
1     1  Negative  ...            3.114808                 4.716881
2     2   Neutral  ...                 NaN                      NaN
3     3   Neutral  ...                 NaN                      NaN
4     4  Negative  ...            1.675922                 2.844206

[4 rows x 7 columns]
```

**Comment by BaggioMarco** [comment]
Hi @zen-juen  thanks for your answer, I try to be more accurate in the description of my goal below.
 
Now let's imagine the following scenario:
I have 3 images "A", "B", "C"  all lasting 6 seconds on the screen and all displayed in row (one after each other).
For the stimulus "B" I detect one scr_peak whose latency (represented by the delay from the stimulus onset to the  SCR_onset ) equal to 1 second.

In this situation I could wonder if such SCR_onset is associated to stimulus "B" or instead it was generated from the observation of the last seconds of exposure to the previous stimulus ("A"). 
This doubt because reading some references, I've seen that latency cannot be treated as contant, instead it can vary around  2-3 seconds. In the some way, I could detect a quick peak in the "C" epoch.. but that could be instead associated to stimulus "B".

MAIN QUESTION related to such scenario:
1) Is it possibile to be sure 100% that a peak occurring after few seconds from the onset of a  stimulus was actually generated from such stimulus and NOT from the previous stimulus? 
From my current understanding I guess that such issue is related to the latency and to the "SCR_RiseTime". 
This because both latency (from stimulus onset to SCR_onset) and "SCR_RiseTime"(from SCR_onset to SCR_peak) cannot be treated as a constant, right? If, instead would be possibile to take into account for each subjects those two time values as constants, it would be really easy solve the issue, and be sure that a specific peak exactly refers to a specific stimulus. 

FURTHER:
2) I've seen that your current "features" report only metrics associated to the first peak, is it correct? What happen if I have more peaks related to the same stimulus, will they be ignored?
3) Finally, could you please to understand the difference between the "SCR_RiseTime" and the "SCR_Peak_Amplitude_Time"? 

Thanks in advance

**Comment by DominiqueMakowski** [comment]
Some quick answers:

> Is it possible to be sure 100% that a peak occurring after few seconds from the onset of a stimulus was actually generated from such stimulus and NOT from the previous stimulus?

No, it is not. There is no way of being sure that a given bodily or brain reaction (and this is not limited to SCRs) is related to a given external stimus. It could be something else entirely. That's why we rely on multiple trials & multiple participants & a priori knowledge about the temporal dynamics of the reactions to mitigate that issue. But it's still an assumption.

> your current "features" report only metrics associated to the first peak, is it correct? What happens if I have more peaks related to the same stimulus, will they be ignored?

That is correct. If you expect multiple peaks, I would run the function several times with varying windows (e.g., 2-7s; 3-7s; 4-7s; 5-7s; etc.). This way you could capture multiple peaks.

**Reply by zen-juen** [reply]
To add on, 
> 2) I've seen that your current "features" report only metrics associated to the first peak, is it correct? What happen if I have more peaks related to the same stimulus, will they be ignored?

You can also use `eda_analyze(data, method="interval-related")` to extract more general characteristics if you have multiple peaks within an epoch, such as number of peaks and the mean amplitude. 

> 3) Finally, could you please to understand the difference between the "SCR_RiseTime" and the "SCR_Peak_Amplitude_Time"?

`RiseTime` refers to the duration that each SCR takes to reach the peak amplitude from onset (indicated by black arrow) and `Peak_Amplitude_Time` is the time point of the first peak *within* each epoch (indicated by blue arrow) - hope the graphic below clarifies.

![image](https://user-images.githubusercontent.com/54053748/131611369-da5669ba-64ee-4dba-b1b8-1b2e4b1842d8.png)

**Comment by BaggioMarco** [comment]
Thank you @DominiqueMakowski  and @zen-juen I really appreciate your comments!
